hyperon_das_atomdb/adapters/hash_table.py

```diff
@@ -619,3 +642,12 @@ def _calculate_composite_type_hash(self, composite_type: list) -> str:
                     _hash_copy
                 )
         return ExpressionHasher.composite_hash(composite_type)
+
+    def _remove_links_not_toplevel(self, matches: list) -> list:
+        matches_copy = matches[:]
```

Comment:
Why copying and then removing non-toplevel instead of creating a new empty list and populate it with toplevel handles? It seems to me that the way you did is a lot slower and odd in terms of design.

Reply:
Yes, you're right. The approach of creating a new list and appending if the condition is better. And it avoids the overhead of copying and removing from the original list. I have not seen. I'm going to change